### ISSUE TITLE
Sly macroexpand functions causing gensyms to be mixed case

### ISSUE THREAD

**Issue by Zulu-Inuoe** [joaotavora/sly#222]
Hello,

It's common as part of a macro to gensym like
```lisp
(let ((sym (gensym (format nil "~A-SOMETHING" existing-sym))))
  ..)
```
However, macroexpanding via the `sly-macroexpand-1-in-place` and others will cause `*print-case*` and others to be set in such a way that I end up with mixed-case-symbols 

This is the slynk function in question:
```lisp
(defun apply-macro-expander (expander string)
  (with-buffer-syntax ()
    (with-bindings *macroexpand-printer-bindings*
      (prin1-to-string (funcall expander (from-string string))))))
```

I think it makes more sense to instead have it do this:
```lisp
(defun apply-macro-expander (expander string)
  (with-buffer-syntax ()
    (let ((expansion (funcall expander (from-string string))))
      (with-bindings *macroexpand-printer-bindings*
        (prin1-to-string expansion)))))
```

This way the expansion itself can use whatever global print settings are active, but the actual expansion print happens with the sly settings

**Comment by joaotavora** [comment]
I'm slightly wary of this because it's such a long standing piece of code, but your reasoning makes sense. I think the reason we haven't gotten bug reports about it is because it is uncommon to use pretty-printing functions in the macro expansion. But it's perfectly legal, so l'll merge it if you make a pull request :)

**Comment by Zulu-Inuoe** [comment]
@joaotavora Yes, I had the thought of 'Well, if case matters the macro should take care to set the right print settings`, but remembering back to common code out there, that's not usually the case.
I suspect nobody complained more due to either not noticing :)

I'll submit the PR, but while we're on this subject:

I'm debugging a complex macro with a compile error in the generated expansion. In order to debug this, I ended up digging a little through sly/slynk and did this:
```lisp
(setf (cdr (assoc '*print-circle* slynk:*slynk-pprint-bindings*)) t)
```
then `sly-macroexpand-1-in-place`

This way I was able to get the macroexpansion in a way I could actually compile (due to generated uninterned symbols) and get a better picture of what was going wrong.

This is definitely not the first time this would have been useful to me, just the first time it bothered me enough for me to dig for it.

I believe that it might be more reasonable to either
1. Have `*print-circle*` set to `t` in slynk's macroexpansion bindings
2. Not bind `*print-circle*` at all so that at least then the global `*print-circle*` setting applies, which is easier to access than the slynk var

Thoughts?

Thanks so much
